playbooks/openshift-glusterfs/upgrade.yml

```diff
@@ -1,4 +1,15 @@
 ---
+- name: Upgrade is in tech preview
+  hosts: all
+  gather_facts: False
+  become: False
+  tasks:
+    - name: Check if user acknowledged that upgrade playbook is in tech preview
+      fail:
+        msg: "Upgrade playbook is in tech preview, please define var openshift_storage_glusterfs_upgrade_in_tech_preview=true to proceed"
+      run_once: True
+      when: openshift_storage_glusterfs_upgrade_in_tech_preview != "true" or openshift_storage_glusterfs_upgrade_in_tech_preview is not defined
```

Comment:
```suggestion
      when: not openshift_storage_glusterfs_upgrade_in_tech_preview | default('true')
```

Reply:
I think it should be default(false) that way if openshift_storage_gluster_update_techpreview is not defined it defaults to false. which is then evaluated to as "when: not false" -> "when: true" which results in the upgrade printing the tech preview warning.

Reply:
one other thing is that now openshift_storage_gluster_update_techpreview can be set to anything for this statement to evaluate as true. which is fine. If the user sets it to something they are acknowledging its in tech preview.

Reply:
@vrutkovs If users set `openshift_storage_glusterfs_upgrade_in_tech_preview` to true and find some issue and then want to disable it, the intuitive thing to do would be to set it to false and then run the playbook again. It will then, not treat the value "false" as any different than "true".

If this is how the rest of the variables are defined in openshift-ansible then I would not mind the change you have suggested, just clarifying what the usual approach is. 

cc: @phlogistonjohn

Reply:
would this be acceptable? it works the same as the statement I previously wrote but is more compact. so openshift_storage_gluster_update_techpreview has to equal true for upgrade to run, any other value or undefined prevents upgrade from running. 

`when:  openshift_storage_gluster_update_techpreview | default('false') != "true"`

Reply:
Ansible would automatically convert `'false'` and `'true'` string into bool, and a shortcut for `!= "true"` is `not openshift_....`

Reply:
Putting "openshift_storage_gluster_update_techpreview=false" in the inventory file still results in the playbook running. I dont think they are being converted to bools. adding " | bool" solves this. 
`when: not openshift_storage_glusterfs_upgrade_in_tech_preview | default('false') | bool`

Reply:
Ah, correct. Looks good now, please squash the commits